components/posts/posts.php

```diff
@@ -253,6 +274,277 @@ function mm_posts_output_masonry_sizers() {
  * @param  object  $context  The global post object.
  * @param  array   $args     The instance args.
  */
+function mm_post_article( $post, $context, $args ) {
+
+	ob_start(); ?>
+
+	<article id="post-<?php the_ID( $post ); ?>" <?php post_class( 'mm-post' ); ?> itemscope itemtype="http://schema.org/BlogPosting" itemprop="blogPost" aria-label="Article">
+
+		<?php do_action( 'mm_posts_header', $post, $context, $args ); ?>
+
+		<?php do_action( 'mm_posts_content', $post, $context, $args ); ?>
+
+		<?php do_action( 'mm_posts_footer', $post, $context, $args ); ?>
+
+	</article>
+
+	<?php return ob_get_clean();
+}
+
+/**
+ * Output a taxonomy term filter.
+ *
+ * @since   1.0.0
+ *
+ * @param   WP_Query  $query    The query object.
+ * @param   WP_Post   $context  The global post object.
+ * @param   array     $args     The instance args.
+ *
+ * @return  string              The category filters HTML.
+ */
+function mm_posts_output_taxonomy_term_filter( $query, $context, $args ) {
+
+	$taxonomy = $args['taxonomy'] ? ( $args['taxonomy'] ) : 'category';
+
+	ob_start(); ?>
+
+	<?php if ( $args['filter_style'] == 'dropdown' ) : ?>
+
+		<div class="mm-posts-filter-wrapper">
+		<div class="mm-posts-filter" >
+		<?php wp_dropdown_categories(array(
+			'taxonomy'         => $taxonomy,
+			'orderby'          => 'name',
+	        'show_option_none' => 'Select',
+			'hide_empty'       => 1,
+	        'value_field'      => 'name',
+			'name'             => 'term-select',
+			'id'               => 'term_dropdown',
+		)); ?>
+
+		</div>
+		</div>
+
+	<?php else: ?>
+
+		<div class="mm-posts-filter-wrapper">
+		<ul class="mm-posts-filter" >
+		<li class="cat-item active"><a href="#" class="mm-posts-filter-all"><?php _e( 'All', 'mm-components' ); ?></a></li>
+		<?php wp_list_categories( array(
+			'title_li'     => '',
+			'hide_empty'   => 1,
+			'hierarchical' => false,
+			'taxonomy'     => $taxonomy,
+			)
+		); ?>
+		</ul>
+		</div>
+
+	<?php endif; ?>
+
+
+	<?php
+
+	return apply_filters( 'mm_posts_taxonomy_term_filter', ob_get_clean(), $query, $context, $args );
+}
+
+/**
+ * Echo the taxonomy term filter.
+ *
+ * @since   1.0.0
+ *
+ * @param   WP_Query  $query    The query object.
+ * @param   WP_Post   $context  The global post object.
+ * @param   array     $args     The instance args.
+ *
+ * @return  string              The category filters HTML.
+ */
+function mm_posts_taxonomy_term_filter( $query, $context, $args ) {
+	echo mm_posts_output_taxonomy_term_filter( $query, $context, $args );
+}
+
+/**
+ * Output taxonomy term filter data attributes.
+ *
+ * @since   1.0.0
+ *
+ * @param   WP_Query  $query    The query object.
+ * @param   WP_Post   $context  The global post object.
+ * @param   array     $args     The instance args.
+ *
+ * @return  string              The category filters HTML.
+ */
+function mm_posts_get_data_attributes( $query, $context, $args ) {
```

Comment:
And actually, now that I think about it, `json_encode()` would make turning the $args into a JS object super easy and allow you DRY the JS data var example up even more.